src/components/Deferred.tsx

```diff
@@ -6,13 +6,7 @@ import {startTransition, use, useLayoutEffect, useState} from 'react';
  * Caution: To achieve performance benefits from this component you have to wrap it in a Suspense component.
  */
 function Deferred({children}: PropsWithChildren) {
-    const [{promise, resolve}] = useState(() => {
-        let resolver!: () => void;
-        const p = new Promise<void>((r) => {
-            resolver = r;
-        });
-        return {promise: p, resolve: resolver};
-    });
+    const [{promise, resolve}] = useState(() => Promise.withResolvers<void>());
```

Comment:
I don't think this should need to be a function?

```suggestion
    const [{promise, resolve}] = useState(Promise.withResolvers<void>());
```

Reply:
Added the polyfill and tested both Android and iOS, both platforms work as intended.